ios/CodePush/CodePushPackage.m

```diff
@@ -292,26 +292,41 @@ + (NSString *)getCodePushPath
 
 + (NSDictionary *)getCurrentPackage:(NSError **)error
 {
-    NSString *packageHash = [CodePushPackage getCurrentPackageHash:error];
-    if (*error || !packageHash) {
+    NSError *localError;
+    NSString *packageHash = [CodePushPackage getCurrentPackageHash:&localError];
+    if (!packageHash) {
+        if (localError && error) {
+            *error = localError;
+        }
         return nil;
     }
-    
-    return [CodePushPackage getPackage:packageHash error:error];
+
+    NSDictionary *package = [CodePushPackage getPackage:packageHash error:&localError];
+    if (!package && localError && error) {
```

Comment:
This seems a little unnecessary (at least to me). Why not just `return [CodePushPackage getPackage:packageHash error:error];`? If `[CodePushPackage getPackage]` deals with the null pointer deference problem too, then that line already achieves all of your goals - 1. To return the result anyway, 2. To only set the `error` pointer pointer if its not nil.